apis/r/configure

```diff
@@ -49,7 +49,7 @@ tools/build_libtiledbsoma.sh
 pkgincl="-I../inst/tiledb/include -I../inst/tiledbsoma/include"
 pkglibs="-ltiledb -L../inst/tiledb/lib -ltiledbsoma -L../inst/tiledbsoma/lib"
 rpath="-Wl,-rpath,'\$\$ORIGIN/../tiledb/lib' -Wl,-rpath,'\$\$ORIGIN/../tiledbsoma/lib'"
-macosver=`${R_HOME}/bin/Rscript -e 'if (Sys.info()["machine"] == "x86_64" && Sys.info()["sysname"] == "Darwin") cat("-mmacosx-version-min=10.15") else cat("")'`
+macosver=`${R_HOME}/bin/Rscript -e 'if (Sys.info()["machine"] == "x86_64" && Sys.info()["sysname"] == "Darwin") cat("-mmacosx-version-min=11.0") else cat("")'`
```

Comment:
We could shorten this because arm64 was always at 11.0, only x86_64 was lower.  So we could shorten this and drop the test for the machine architecture and only condition on Darwin.

```suggestion
macosver=`${R_HOME}/bin/Rscript -e 'if (Sys.info()["sysname"] == "Darwin") cat("-mmacosx-version-min=11.0") else cat("")'`
```